unified_planning/io/anml_writer.py

```diff
@@ -0,0 +1,329 @@
+        return f'({args[0]} iff {args[1]})'
+
+    def walk_fluent_exp(self, expression, args):
+        fluent = expression.fluent()
+        if len(args) == 0:
+            return fluent.name
+        else:
+            return f'{fluent.name}({", ".join(args)})'
+
+    def walk_param_exp(self, expression, args):
+        assert len(args) == 0
+        return expression.parameter().name
+
+    def walk_object_exp(self, expression, args):
+        assert len(args) == 0
+        return expression.object().name
+
+    def walk_bool_constant(self, expression, args):
+        assert len(args) == 0
+        if expression.bool_constant_value():
+            return 'true'
+        return 'false'
+
+    def walk_real_constant(self, expression, args):#NOTE is this right? Same ANML Implementation
+        assert len(args) == 0
+        frac = expression.constant_value()
+
+        with localcontext() as ctx:
+            ctx.prec = self.DECIMAL_PRECISION
+            dec = frac.numerator / Decimal(frac.denominator, ctx)
+
+            if Fraction(dec) != frac:
+                warn("The ANML printer cannot exactly represent the real constant '%s'" % frac)
+            return str(dec)
+
+    def walk_int_constant(self, expression, args):
+        assert len(args) == 0
+        return str(expression.constant_value())
+
+    def walk_plus(self, expression, args):
+        assert len(args) > 1
+        return f"({' + '.join(args)})"
+
+    def walk_minus(self, expression, args):
+        assert len(args) == 2
+        return f'({args[0]} - {args[1]})'
+
+    def walk_times(self, expression, args):
+        assert len(args) > 1
+        return f"({' * '.join(args)})"
+
+    def walk_div(self, expression, args):
+        assert len(args) == 2
+        return f'({args[0]} / {args[1]})'
+
+    def walk_le(self, expression, args):
+        assert len(args) == 2
+        return f'({args[0]} <= {args[1]})'
+
+    def walk_lt(self, expression, args):
+        assert len(args) == 2
+        return f'({args[0]} < {args[1]})'
+
+    def walk_equals(self, expression, args):
+        assert len(args) == 2
+        return f'({args[0]} == {args[1]})'
+
+
+class ANMLWriter:
+    '''This class can be used to write a Problem in ANML.'''
+
+    def __init__(self, problem: 'up.model.Problem'):
+        self.problem = problem
+
+    def _write_problem(self, out: IO[str]):
+        types_mapping: Dict['up.model.Type', str] = {}
+        # Init types_mapping.
+        types_mapping[self.problem.env.type_manager.IntType()] = 'integer'
+        types_mapping[self.problem.env.type_manager.RealType()] = 'rational'
+        for t in self.problem.user_types:
+            ut = cast(_UserType, t)
+            if _is_valid_anml_name(ut.name): # No renaming needed
+                types_mapping[t] = ut.name
+
+        for t in self.problem.all_types:
+            #NOTE If ANML does not accept a re-defintion of 'integer' or 'rational' or maybe also 'boolean', they must be skipped
```

Comment:
Not sure to understand the note: `integer`, `float`, `rational`, `boolean` are reserved keywords in ANML...